### DISCUSSION TITLE
Does QuestDB support percentile aggregation?

### DISCUSSION THREAD

**Discussion by pickledish** [questdb/questdb#1135] [Ideas]
I've looked in the docs and wasn't able to find anything aside from the regular aggregations (min, avg, etc), and playing with Postgres syntax for it on `demo.questdb.io` suggests that it isn't supported.

Are there plans for this? Could look into something like https://github.com/tdunning/t-digest where intermediate results can be partitioned and merged

**Comment by bluestreak01** [comment]
We can certainly have a look into this if you have interesting usecase for questdb. Our codebase is dependency free, which is why adding t-digest algorithms is a little more complicated affair than using a library. 

Thank you for the pointers though! I am not familar with the algorithm personally but it looks very curious. These are useful papers on the subject:

https://github.com/tdunning/t-digest/blob/master/docs/t-digest-paper/histo.pdf

and impleentation is Facebook folly:

https://github.com/facebook/folly/blob/master/folly/stats/TDigest.h

We should try to understand the algorigth and see if it lends itself to SIMD processing.

**Comment by shimondoodkin** [comment]
i wanted this too

**Reply by pswu11** [reply]
Hi, what is your use case? 
There's a related issue https://github.com/questdb/questdb/issues/1371 
feel free to upvote/comment if that also suits your use case :)

**Reply by shimondoodkin** [reply]
upvoted,

i wanted a value by frequency.  i wanted a regular percentile on a value. 
kind of median, maybe little higher value but without outliers of short tail.

**Comment by puzpuzpuz** [comment]
Related issue: https://github.com/questdb/questdb/issues/1371
